### PR TITLE
chore(CPlusPlus): add return string subsequences

### PR DESCRIPTION
**Pull request by vivekthedev**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [ ] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Return String Subsequences using Recursion

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Recursion/string-subsequences.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Appraoch:
+Subsequences are the every possible sequences that can be formed from a string
+from a empty string to every character in the string.
+
+In the below code we return the array size which contains our subsequences.
+In every recursion call the function get the double subsequences of what it was
+before by adding single character to the previous call.
+*/
+
+#include <iostream>
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Recursion/string-subsequences.cpp:17]:
```suggestion
#include <iostream>
#include <string>
```

Make sure to include all the required headers, even though some implementations include string header which implementing parts of iostream, but they are not obligated to do so.

File: algorithms/CPlusPlus/Recursion/string-subsequences.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        output[i + smallOutputSize] = input[0] + output[i];
+    }
+    return 2 * smallOutputSize;
+}
+
+int main()
+{
+    string s = "abc";
+    string *out = new string[1000];
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Recursion/string-subsequences.cpp:38]: what if the number of substrings is more than 1000 ?

An alternative here would be to use a flexible container ( like std::vector ) and let it do the heavy lifting of memory management while you take care of this program.

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]

**Comment by Arsenic-ATG** [comment]
Along with changes suggested, make sure you also update the required README.md index of Cplusplus dir with your creation in it.

**Comment by ming-tsai** [comment]
Hi @vivekthedev,
Any update?

**Pull request closed without merging**